File: 231-PowerOfTwo/231-PowerOfTwo.py
# Last updated: 8/13/2025, 8:20:54 PM
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n==1 :
            return True
        
        elif n%2!=0 or n==0 :
            return False
        
        else :
            return self.isPowerOfTwo(n/2)
        
    

File: 231-PowerOfTwo/231-PowerOfTwo.java
// Last updated: 8/13/2025, 8:21:02 PM
class Solution {
    public boolean isPowerOfTwo(int n) {
        if(n==1){
            return true;
        }
        else if(n%2!=0 || n==0){
            return false;
        }
        else{
            return isPowerOfTwo(n/2);
        }
    }
}